app/scripts/modules/amazon/src/loadBalancer/details/loadBalancerDetails.controller.ts

```diff
@@ -0,0 +1,220 @@
+      return detailsLoader.then((details) => {
+        this.loadBalancer = appLoadBalancer;
+        this.state.loading = false;
+        const securityGroups: IApplicationSecurityGroup[] = [];
+        if (details.length) {
+          this.loadBalancer.elb = details[0] as IAmazonLoadBalancer;
+          this.loadBalancer.account = this.loadBalancerFromParams.accountId;
+
+          if (details[0].loadBalancerType === 'application') {
+            const elb = details[0] as IAmazonApplicationLoadBalancer;
+            if (elb.listeners && elb.listeners.length) {
+              this.elbProtocol = 'http:';
+              if (elb.listeners.some((l: any) => l.protocol === 'HTTPS')) {
+                this.elbProtocol = 'https:';
+              }
+
+              this.listeners = [];
+              elb.listeners.forEach((listener) => {
+                listener.defaultActions.forEach((action) => {
+                  const targetGroup = (this.loadBalancer as IAmazonApplicationLoadBalancer).targetGroups.find((tg) => tg.name === action.targetGroupName) || { name: action.targetGroupName } as ITargetGroup;
+                  this.listeners.push({ in: `${listener.protocol}:${listener.port}`, target: targetGroup });
+                })
+              })
+            }
+          } else {
+            // Classic
+            if (details[0].listenerDescriptions) {
+              this.elbProtocol = 'http:';
+              if (details[0].listenerDescriptions.some((l: any) => l.listener.protocol === 'HTTPS')) {
+                this.elbProtocol = 'https:';
+              }
+            }
+          }
+
+          this.loadBalancer.elb.securityGroups.forEach((securityGroupId: string) => {
+            const match = this.securityGroupReader.getApplicationSecurityGroup(this.app, this.loadBalancerFromParams.accountId, this.loadBalancerFromParams.region, securityGroupId);
+            if (match) {
+              securityGroups.push(match);
+            }
+          });
+          this.securityGroups = sortBy(securityGroups, 'name');
+
+          if (this.loadBalancer.subnets) {
+            this.loadBalancer.subnetDetails = this.loadBalancer.subnets.reduce((subnetDetails: ISubnet[], subnetId: string) => {
+              this.subnetReader.getSubnetByIdAndProvider(subnetId, this.loadBalancer.provider)
+                .then((subnetDetail: ISubnet) => {
+                  subnetDetails.push(subnetDetail);
+                });
+
+              return subnetDetails;
+            }, []);
+          }
+        }
+      },
+        () => this.autoClose()
+      );
+    } else {
+      this.autoClose();
+    }
+    if (!this.loadBalancer) {
+      this.autoClose();
+    }
+
+    return this.$q.when(null);
+  }
+
+  public getFirstSubnetPurpose(subnetDetailsList: ISubnet[] = []) {
+    return head(subnetDetailsList.map(subnet => subnet.purpose)) || '';
+  };
+}
+
+
+export const AWS_LOAD_BALANCER_DETAILS_CTRL = 'spinnaker.loadBalancer.aws.details.controller';
```

Comment:
Yep, easy. Also: We should standardize on one or the other, there's aws and amazon all over the place.